File: Old Code/WurmRancher/WurmRancher/PoisonWeed.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Windows.Media.Imaging;

namespace WurmRancher
{
    public class PoisonWeed :  EdiblePlant
    {
        static BitmapImage PoisonWeedBitmap = new BitmapImage();
        static BitmapImage PoisonWeedSprayedBitmap = new BitmapImage();
        static Random rand = new Random();
        static PoisonWeed()
        {            
            PoisonWeedBitmap.SetSource(WRResources.poison_weed);
            PoisonWeedSprayedBitmap.SetSource(WRResources.poison_weed_sprayed);
        }
        
        public PoisonWeed(GameControl theControl_, int indexX, int indexY) : base(theControl_, indexX, indexY) 
        {
            this.PieceImage.Source = PoisonWeedBitmap;
            eat_value = -1;
        }

        int count = 0;
        //int spawn_rate = 500;
        public override void Update()
        {
            count++;
            if (count >= theControl.CurrentLevel.PoisonWeedGrowthRate && !IsSprayed)
            {
                count = 0;
                switch (rand.Next(4))
                {
                    case 0:
                        theControl.GrowPoisonWeed(IndexX - 1, IndexY);
                        break;
                    case 1:
                        theControl.GrowPoisonWeed(IndexX + 1, IndexY);
                        break;
                    case 2:
                        theControl.GrowPoisonWeed(IndexX, IndexY - 1);
                        break;
                    case 3:
                        theControl.GrowPoisonWeed(IndexX, IndexY + 1);
                        break;
                }
            }
            base.Update();
           
        }

        protected override BitmapImage SprayedPicSource
        {
            get { return PoisonWeedSprayedBitmap; }
        }


        public override bool Available
        {
            get { return !this.sprayed; }
        }
       
    }
}

File: ts/poisonWeed.ts
import { EdiblePlant } from "./ediblePlant.js";
import { CurrentLevel, GrowPoisonWeed } from "./gameControl.js";
import { GrassChaser } from "./predPrey.js";
import { poisonWeedImage, poisonWeedSprayedImage } from "./resources.js";

export class PoisonWeed extends EdiblePlant{

    SprayedPic = poisonWeedSprayedImage;
    Name = "PoisonWeed";

    constructor(indexX:number, indexY:number){
        super(indexX,indexY);
        this.PieceImage = poisonWeedImage;

            this.eat_value = -1;
    }

    count = 0;
    Update(time_step:number):void{
        this.count+=time_step;
        if (this.count >= CurrentLevel.WeedGrowthRate && !this.sprayed)
        {
            this.count = 0;
            let r = Math.random();
            if (r<.25)
                GrowPoisonWeed(this.indexX - 1, this.indexY);
            else if (r < .5)
                GrowPoisonWeed(this.indexX + 1, this.indexY);
            else if (r < .75)
                GrowPoisonWeed(this.indexX,  this.indexY-1);
            else
                GrowPoisonWeed(this.indexX, this.indexY+1);
        }
        super.Update(time_step);
    }

    Available(eater : GrassChaser): boolean {
        return (!this.sprayed) && super.Available(eater);
    }
}